src/arch/xtensa/CMakeLists.txt

```
@@ -135,7 +139,7 @@ target_compile_options(sof_options INTERFACE
 		${XTENSA_C_FLAGS}
 		${EXTRA_CFLAGS_AS_LIST}
 	>
-	-imacros${CONFIG_H_PATH}
+	-imacros ${CONFIG_H_PATH}
```

From line 130: "No space between -imacros and its argument to avoid CMake de-duplication "feature"

---

@marc-hb yeah... is it still relevant / important? Sparse cannot handle this with no space :-/ Seems to work like this too

---

Maybe not anymore, in that case the comment must be changed. If you start with git blame you will find discussions and links about this.

---

Ping? I vaguely remember some Zephyr updates about this but what about XTOS? I don't remember all the details but I remember this single whitespace took me hours of research and testing to figure out, so "seem to work now" is not enough sorry.

---

@lyakh you should be using `SHELL:` here:
https://cmake.org/cmake/help/latest/command/target_compile_options.html#option-de-duplication
```suggestion
	"SHELL:-imacros ${CONFIG_H_PATH}"
```

to avoid de-duplication of `-imacros` in case more users / targets adds `-imacros <something>` to any target using the `sof_options` interface lib.